src/TLuaInterpreter.cpp

```diff
@@ -9009,16 +9009,63 @@ int TLuaInterpreter::createMapLabel(lua_State* L)
 int TLuaInterpreter::setMapZoom(lua_State* L)
 {
     qreal zoom = getVerifiedFloat(L, __func__, 1, "zoom");
+    std::optional<int> areaID;
+    if (lua_gettop(L) > 1) {
+        areaID = getVerifiedInt(L, __func__, 2, "area id", true);
+    }
     Host& host = getHostFromLua(L);
-    if (host.mpMap) {
-        if (host.mpMap->mpMapper) {
-            if (host.mpMap->mpMapper->mp2dMap) {
-                host.mpMap->mpMapper->mp2dMap->setMapZoom(zoom);
-                updateMap(L);
-            }
+    if (host.mpMap.isNull() || host.mpMap->mpMapper.isNull()) {
+        return warnArgumentValue(L, __func__, "no map loaded or no active mapper");
+    }
+
+    if (zoom < T2DMap::csmMinXYZoom) {
+        return warnArgumentValue(L, __func__, qsl("%1 is less than the minimum value allowed for zoom which is %2").arg(QString::number(zoom), QString::number(T2DMap::csmMinXYZoom)));
+    }
+
+    bool result = false;
+    if (areaID.has_value()) {
+        result = host.mpMap->mpMapper->mp2dMap->setMapZoom(areaID.value(), zoom);
+    } else {
+        host.mpMap->mpMapper->mp2dMap->setMapZoom(zoom);
+        result = true;
+    }
+
+    if (!result) {
+        // Whilst it will also return false should there not be a map or mapper
+        // we should have already ruled those cases out:
+        return warnArgumentValue(L, __func__, qsl("number %1 is not a valid area id").arg(QString::number(areaID.value())));
```

Comment:
as discussed in https://github.com/Mudlet/Mudlet/issues/2605
```suggestion
        return warnArgumentValue(L, __func__, qsl("number %1 is not a valid areaID").arg(QString::number(areaID.value())));
```

Reply:
There are multiple instances of *both* texts in the file IIRC and I didn't do a count of each to find which one was precisely the most common... as it is that issue has not, um, been actioned/resolved yet! However I will take on board that you have pointed it out in regard to these texts in the PR.